components/common/FloatingRibbon.tsx

```diff
@@ -19,7 +20,13 @@ export default function FloatingRibbon({ children }) {
   )
 }
 
-export function Button({ children, ...props }) {
+export function Button({
+  children,
+  ...props
+}: {
+  children: ReactNode
+  [x: string]: unknown
+}): ReactElement {
```

Comment:
I find this 7-line, multiply indented signature for a very basic component pretty hard to read. Maybe you get used to it as you get more used to TypeScript, but my suggestion would be to define a `Props` type separately:

```
interface Props {  // or type?
  children: ReactNode
  [x: string]: unknown
}

export function Button({ children, ...props }: Props) {
```

Do others find this easier to read? Maybe it would be a better convention to follow?

Reply:
Agreed. I'd prefer to see our types/interfaces defined at the top of a file below `imports`

Reply:
Totally get what both of you are saying, I was trying to keep things the same across everything, and for a lot of them they are single prop, so it felt weird to have a fully separated type/interface above it, but then for these larger ones I definitely agree it becomes a mess. I don't like having 2 standards for where things are/defined, it just leads to messiness down the road. Will refactor them though, I agree that for the larger ones its just a mess

Reply:
I'm sorry I missed this before, but I agree with you about the simple, single-prop ones. This does feel a bit like overkill. I think my preference would be to allow different styles for short, simple ones vs. long, complicated ones. I'd probably put the boundary at one line: If doing it inline would make the function signature multi-line (because of the number of complexity of the props), then factor it out into a standalone interface.

That said, if you prefer to keep it consistent or you just don't want to change it now after all the work you've done, I'm 100% supportive of that choice.

Reply:
Yeah, I would agree and I think I'll make one last tweak in that vein. Gonna kinda go by the following rules of if its one prop do it inline, otherwise a separate block. I find even with 2, it gets messy looking inline, so it makes more sense to do a quick interface above

Reply:
I've gone back and forth mentally on this now a few times. I think I want to keep it consistent as well as always do the interface. doing the inline when there is destructuring makes it end up looking like you are doing destructuring default assignments, which confuses quick reading, where as leaving the interface separate makes it clearer!